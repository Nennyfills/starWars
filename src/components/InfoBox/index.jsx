import React from 'react';
import PropTypes from 'prop-types';

import './infoBox.scss';

/**
 * InfoBox Component
 *  @param {string} message
 *  @returns infoBox element
 */
const InfoBox = ({ message }) => (
  <div className="infoBox">
    <div className="error">
      <i className="fa fa-times-circle" />
      {message}
    </div>
  </div>
);

InfoBox.propTypes = {
  message: PropTypes.string,
};
InfoBox.defaultProps = {
  message: '',
};
export default InfoBox;
