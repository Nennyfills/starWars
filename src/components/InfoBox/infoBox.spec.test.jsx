import React from 'react';

import InfoBox from './index';


describe('InfoBox Component', () => {
  it('should mount the InfoBox wrapper', () => {
    const wrapper = mount(<InfoBox />);

    expect(wrapper.find('.infoBox')).toBeDefined();
    expect(wrapper.find('.error')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
