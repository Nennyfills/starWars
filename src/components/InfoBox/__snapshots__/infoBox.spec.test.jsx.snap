// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox Component should mount the InfoBox wrapper 1`] = `
<InfoBox>
  <div
    className="infoBox"
  >
    <div
      className="error"
    >
      <i
        className="fa fa-times-circle"
      />
    </div>
  </div>
</InfoBox>
`;
